pkg/distributor/distributor.go

```diff
@@ -1099,15 +1093,22 @@ func (d *Distributor) prePushRelabelMiddleware(next PushFunc) PushFunc {
 			return err
 		}
 
+		validationScheme := d.limits.ValidationScheme(userID)
+
 		var removeTsIndexes []int
 		lb := labels.NewBuilder(labels.EmptyLabels())
 		for tsIdx := 0; tsIdx < len(req.Timeseries); tsIdx++ {
 			ts := req.Timeseries[tsIdx]
 
-			if mrc := d.limits.MetricRelabelConfigs(userID); len(mrc) > 0 {
+			if mrcs := d.limits.MetricRelabelConfigs(userID); len(mrcs) > 0 {
+				for _, mrc := range mrcs {
```

Comment:
this currently runs for every incoming sample; can we make a change somewhere where the relabel configs are loaded so we set the validation scheme once instead of on every ingested sample for every relabel config?

Reply:
Was thinking about this when implementing. I would place this check in util/limits.validate, but the validation scheme doesn't have a yaml parameter (yet). 

I moved it up so it's not run for each series (but still for each sample). Do you have any other suggestion? @aknuds1